ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java

```diff
@@ -131,6 +125,11 @@ public String getKafkaTopic() {
     return Optional.ofNullable(props.getInt(CreateConfigs.SCHEMA_ID));
   }
 
+  public Optional<FormatInfo> getKeyFormatInfo() {
+    return Optional.ofNullable(props.getString(CommonCreateConfigs.KEY_FORMAT_PROPERTY))
+        .map(format -> FormatInfo.of(format, ImmutableMap.of()));
+  }
```

Comment:
This should also return non-empty if `FORMATS` is set, (same for value version).